### PR TITLE
feat: wrap subQuery with parenthesis when it appears as table name

### LINKED ISSUE #3485: please hint about missing "as()" when from(subquery) is missing it
# Environment

Knex version: 0.19.5
Database + version: postgresql 11
OS: linux/debian

# Feature discussion / request

I find myself often spending hours trying to figure how to make
```
knex('table').select(rawstr).from(knex('table').where(obj))
```
work.
It's very simple:
```
knex('table').select(rawstr).from(knex('table').where(obj).as('sub'))
```

However knex outputs a wrong sql statement (missing parenthesis) and when debugging, it looks like a bug in knex.

I think it would be nice to users if a missing as() could be detected and a warning printed.

### PR DESCRIPTION
**Pull request by edvardchen**
Fixes #3485

### CODE REVIEW

**Comment by kibertoad** [comment]
Could you please also add integration test that would actually execute the generated sql?

File: test/integration/builder/selects.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /*global expect, d*/
 'use strict';
 
+const mysqlErrors = require('mysql/lib/protocol/constants/errors');
```
Comment (kibertoad) [inline test/integration/builder/selects.js]: Why depend on non-public part of external dependency and not just hardcode the constant internally?

Comment (edvardchen) [reply test/integration/builder/selects.js]: Emm, OK. I will change it

Comment (kibertoad) [reply test/integration/builder/selects.js]: My concern is that if there is a major change in the driver and both the constant is no longer there and field is no longer set, comparison is still going to be "undefined === undefined" and will pass. Thanks for changing it!

**Comment by kibertoad** [comment]
Thanks!

**Pull request merged**